### ISSUE TITLE
Recent S3MultiRegionClient changes caused issue

### ISSUE THREAD

**Issue by jeremeamia** [aws/aws-sdk-php#1324]
Hello PHP team,

I didn't see any changes to the [`S3MultiRegionClient`](https://github.com/aws/aws-sdk-php/blob/master/src/S3/S3MultiRegionClient.php) called out in any release notes, but it looks like some fairly extensive work was done for 3.25.0. After a recent deployment, we started seeing some issues like:
```
Error executing "PutObject" on "https://{BUCKET}/{OBJECT_KEY}"; AWS HTTP error: Client error: `PUT https://{BUCKET}/{OBJECT_KEY}` resulted in a `400 Bad Request` response:
    AuthorizationHeaderMalformed
    The authorization header is malformed; the region 'us-east-1' is wrong; expecting 'us-west-2'
    {SIGNATURE_STUFF}
```
But we didn't change any code or config other than the SDK version. So, would any of the recent changes to `S3MultiRegionClient` effect already running code? (e.g., changes to caching, etc.)

**Comment by imshashank** [comment]
@jeremeamia The last update to the S3MultiRegionClient was done in this PR -> https://github.com/aws/aws-sdk-php/pull/1231. The code update was in response to a bug. 

Looks like from your comment that this doesn't happen on each request. Please correct me if I am wrong.

Also, it will be really helpful is you can provide me with any information helpful in reproducing the bug.

**Comment by jeremeamia** [comment]
> Looks like from your comment that this doesn't happen on each request. Please correct me if I am wrong.

It happens on every request to buckets in us-west-2. The versions of our app interacting exclusively with buckets in us-east-1 still work.

We've downgraded to version "~3.24.0" for the time being.

SDK is configured like this:
```php
$aws = new Sdk([
    'region' => 'us-east-1',
    'version' => 'latest',
    'credentials' => [
        'key' => getenv('AWS_KEY'),
        'secret' => getenv('AWS_SECRET'),
    ],
    'S3' => [
        'bucket_region_cache' => $cache,
    ],
]);
// ...
$s3 = $aws->createMultiRegionS3();
// ...
$s3->putObject([
    // standard stuff (e.g., Bucket, Key, SourceFile, ServerSideEncryption, ACL)
]);
```

The code and config has been like this for almost a year, the only thing that has changed is the SDK version.

**Comment by jeremeamia** [comment]
Tagging @Azuka, in case he wants to follow the conversation or add details.

**Comment by imshashank** [comment]
@jeremeamia Thanks for the update. Working on figuring out the issue.

**Comment by imshashank** [comment]
@jeremeamia We are currently investigating the issue. We have figured out that the error doesn't occur if you configure the client with `use_path_style_endpoint`.

Will push a bug fix very soon.

**Comment by almasry** [comment]
+1

**Comment by durzo** [comment]
+1

**Comment by kstich** [comment]
Posted a fix that attempts to read the passed `<Region>` out of the body of the `AuthorizationHeaderMalformed` response, or proceeds to make another request to determine the correct region (emulating a 301 with `x-amz-bucket-region`.)

**Comment by kyrozetera** [comment]
I just ran into this as well. Another workaround is not to configure a default region in the S3 client. In this case it doesn't throw the exception. However, this requires that you don't pass this config to the `Sdk` constructor, but configure the specific clients in their factory method or constructor instead.

**Comment by kstich** [comment]
The fix from #1330 was added to the release of [3.31.7](https://github.com/aws/aws-sdk-php/releases/tag/3.31.7). Thanks everyone for reporting this issue!
